src/IO.go

```diff
@@ -2,31 +2,43 @@ package main
 
 import (
 	"fmt"
+	"io"
 	"os"
 	"strings"
+
+	log "github.com/Sirupsen/logrus"
 )
 
-func formatAndPrint(planets []Planet, opts *Opts) {
-	formatter := Formatter{}
-	failed := false
+func formatAndPrint(planets []Planet, opts *Opts, writer io.Writer) {
+	factory := Formatter{}
+	formatter := factory.getFormatter(opts)
+	if formatter == nil {
+		printUnformatted(planets, writer)
+		return
+	}
+
+	log.Debugf("using formatter of type : %T", formatter)
 	formatter.init()
-	var formatted string
+	formatter.format(planets, opts, writer)
+
 	for _, entry := range planets {
 		if entry.outputStruct.errored {
-			failed = true
+			os.Exit(1)
```

Comment:
Dies ist die einzige Stelle wo es geschieht. Im master war es drin ich habe es einfach uebernommen. Diese Anweisung ist  wie man sieht am Ende der Ausfuehrung.
Ist vielleicht Ueberfluessig. Wuerde die Frage aber gerne in einem anderen Ticket naeher untersuchen.